lib/logstash/outputs/config/bigdecimal_patch.rb

```diff
@@ -0,0 +1,24 @@
+require 'bigdecimal'
+
+class BigDecimal
+    # Floating-point numbers that go through the 'json' Logstash filter get automatically converted into BigDecimals.
+    # Example of such a filter:
+    #
+    # filter {
+    #   json {
+    #     source => "message"
+    #   }
+    # }
+    #
+    # The problem is that { "value" => BigDecimal('0.12345') } gets serialized into { "value": "0.12345e0"}. We do
+    # want to keep floating point numbers serialized as floating point numbers, even at the expense of loosing a little
+    # bit of precision during the conversion. So, in the above example, the correct serialization would be:
+    # { "value": 0.12345}
+    def to_json(options = nil) #:nodoc:
+      if finite?
+        self.to_f.to_s
+      else
+        'null'
```

Comment:
what does BigDecimal.to_json do in this case? Why not `NaN` or something? (I have no idea! 😛 )

Reply:
This statement means that when trying to encode `BigDecimal('NaN')` in JSON, it will be encoded as 'null'. The JSON encoder treats 'null' actually as a real `nil` value, as you can see in the following test: https://github.com/newrelic/logstash-output-plugin/blob/77c09176f80509af013e23d26889b981e2276fa0/spec/outputs/newrelic_spec.rb#L316-L322